### DISCUSSION TITLE
Hi I am having errors while using SINDy-PI, i am using for implicit ODE

### DISCUSSION THREAD

**Discussion by Subodh213** [dynamicslab/pysindy#487] [Q&A] [answered]
Hi, I am having issue running SINDu Pi for implicit ODE, someone suggest it is due to sci-kit version , so I downgraded it to lower version but I am still having same error: 

AttributeError                            Traceback (most recent call last)
Cell In[3], line 50
     41 # Need to pass time base to the library so can build the x_dot library from x
     42 sindy_library = ps.SINDyPILibrary(
     43     library_functions=library_functions,
     44     x_dot_library_functions=x_dot_library_functions,
   (...)
     47     include_bias=True,
     48 )
---> 50 sindy_opt = ps.SINDyPI(
     51     threshold=1e-6,
     52     tol=1e-8,
     53     thresholder="l1",
     54     max_iter=20000,
     55 )
     57 model = ps.SINDy(
     58     optimizer=sindy_opt,
     59     feature_library=sindy_library,
     60     differentiation_method=ps.FiniteDifference(drop_endpoints=True),
     61 )
     62 model.fit(x_train, t=t)

AttributeError: module 'pysindy' has no attribute 'SINDyPI'

Kindly suggest how to fix this

**Comment by Jacob-Stevens-Haas** [comment, marked as answer]
It is likely that you did not install the cvxpy extras.  Run 

`pip install pysindy[cvxpy]` next time to get `scs` and `cvxpy` packages.  These are required for the SINDyPI optimizer.  From your current position, you can just try to run `pip install scs cvxpy`.

**Reply by Subodh213** [reply]
Thanks it worked!
